### ISSUE TITLE
Error about meta_optimizer and new data

### ISSUE THREAD

**Issue by leyi-123** [HLTCHKUST/PAML#15]
Hello! I use my own data to train your model. After line 170 `meta_optimizer.step()` is executed, line 150 `val_loss, v_ppl = do_learning_fix_step(meta_net, train_iter, val_iter, iterations=config.meta_iteration)` outputs `val_loss` as `tensor(nan, device='cuda:0', grad_fn=<AddBackward>)`, which causes the training failure. I didn't change your code but persona_map, and I want to know what went wrong. Thanks!

**Comment by andreamad8** [comment]
Maybe you need to provide a little bit more details. 

This could happen for many reasons:
- the train_iter or val_iter is empty
- the lr is too high
or more.

**Comment by leyi-123** [comment]
> 
> 
> Maybe you need to provide a little bit more details.
> 
> This could happen for many reasons:
> 
>     * the train_iter or val_iter is empty
> 
>     * the lr is too high
>       or more.
I'm sorry that my question may be too sketchy. Now together with my last question, I will tell you in detail! I have trained on your model with my own data, which is in the same format as the example you gave, as follows:
![0c8950903140cefc161f24644d5a307](https://user-images.githubusercontent.com/62412393/102714040-37818800-4307-11eb-9f58-477dbc59389a.png)

The persona part is not a description, but a person's ID. I changed the `cluster_persona` function in `data_reader.py` as follows:
![336275b2406bbfb77a20d29fef0a15f](https://user-images.githubusercontent.com/62412393/102714075-61d34580-4307-11eb-815c-74f4655326c1.png)

 and `persona_map.txt` is just like:
![0db2db71de763ca935f4e738dc75376](https://user-images.githubusercontent.com/62412393/102714157-b8408400-4307-11eb-8a4e-b545012bbc1d.png)

When I ran `MAML.py`, I found that I couldn't train the model I wanted. After observation, I found that the functions, `do_evaluation` (defined on line 96 in `MAML.py`) and `do_learning_fix_step` (defined on line 74 in `MAML.py`), both returned `tensor(nan, device='cuda:0', grad_fn=<AddBackward>) `after the `meta_optimizer.step()` (line 170), which led to the training failure. So I want to know how to solve it. Thank you very much!

**Comment by andreamad8** [comment]
mmm I see. I really don't know at this point. 

I suggest going step by step inside the do_evaluation function to check where the loss gets none. 

Sorry I cannot help much here.
